cli/src/main.rs

```diff
@@ -52,6 +52,225 @@ pub fn main() {
         }
     };
 
+    let regtest = matches.is_present("regtest");
+    if regtest {
+        use std::ffi::OsString;
+        use std::fs::File;
+        use std::io;
+        use std::path::PathBuf;
+        use std::process::{Command, Stdio};
+        use std::{thread, time};
+
+        // confirm we are in running from zingolib as an anchor for our directory context
+        let revparse = Command::new("git")
+            .args(["rev-parse", "--show-toplevel"])
+            .output()
+            .expect("no git!? time to quit.");
+
+        let revparse_str = std::str::from_utf8(&revparse.stdout).expect("revparse_str error");
+
+        //if revparse ends in zingolib, use that as the start to build the dir
+        if revparse_str.trim_end().ends_with("zingolib") {
+            // proceed
+        } else {
+            panic!("Zingo-cli's regtest mode must be run within its git tree");
+        }
+
+        // Cross-platform OsString
+        let mut bin_directions = OsString::new();
+        bin_directions.push(revparse_str.trim_end());
+
+        // convert this back into a path for windows compatibile dir building
+        let bin_pathbuf: PathBuf = [
+            bin_directions.clone(),
+            OsString::from("regtest"),
+            OsString::from("bin"),
+        ]
+        .iter()
+        .collect();
+
+        let zcash_conf_pathbuf: PathBuf = [
```

Comment:
Let's remove all type-hints.   That is, no var `ident` should contain a type "name".

Specifically, rename `zcash_conf_pathbuf` to `zcash_confs`